<?xml version="1.0"?>
<!--
Copyright (C) 2020, Raffaello Bonghi <raffaello@rnext.it>
All rights reserved
Redistribution and use in source and binary forms, with or without
modification, are permitted provided that the following conditions are met:
1. Redistributions of source code must retain the above copyright 
   notice, this list of conditions and the following disclaimer.
2. Redistributions in binary form must reproduce the above copyright
   notice, this list of conditions and the following disclaimer in the
   documentation and/or other materials provided with the distribution.
3. Neither the name of the copyright holder nor the names of its 
   contributors may be used to endorse or promote products derived 
   from this software without specific prior written permission.
 
THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND 
CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, 
BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS 
FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, 
SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; 
OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, 
WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE 
OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, 
EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-->
<launch>
  <!-- Run ROS jetson stats with default configuration -->
  <arg name="port"        default="/dev/ttyUSB0" />
  <arg name="config"      default="$(find roboteq_control)/config/roboteq.yaml" />
  <arg name="respawn"     default="false" />
  <arg name="output"      default="log" />
  <!-- Machine name if default run in localhost -->
  <arg name="machine"         default="_default_" />
  
  <group if="$(eval arg('machine') == '_default_')">
    <machine name="_default_" address="localhost" default="never" timeout="10"/>
  </group>

  <!-- Roboteq hardware driver -->
  <node machine="$(arg machine)" pkg="roboteq_control" type="driver_node" name="roboteq"
          output="$(arg output)" respawn="$(arg respawn)" respawn_delay="60">
    <rosparam command="load" file="$(arg config)" />
    <rosparam subst_value="true">
      serial_port: $(arg port)
      serial_rate: 115200
      control_frequency: 50.0
      diagnostic_frequency: 5.0
    </rosparam>
  </node>

</launch>
